File: app/src/main/java/com/tarighi/register/UserController.java
package com.tarighi.register;

import android.util.Log;

import com.orhanobut.hawk.Hawk;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

public final class UserController {
    private static List<UserInfo> Users;
    public static final UserController INSTANCE;

    static {
        INSTANCE = new UserController();
    }

    private UserController() {
        init();
    }

    @Nullable
    public final List<UserInfo> getUsers() {
        if (!Hawk.contains(Constant.USERLIST)) {
            Users = new ArrayList<UserInfo>();
        }
        Users = Hawk.get(Constant.USERLIST);
        if (Users == null) {
            Users = new ArrayList<UserInfo>();
        }
        return Users;
    }

    private final void setUsers(List<UserInfo> users) {
        Users = users;
    }

    private final void addUser(UserInfo user) {
        user.Id = Users.size() + 1;
        Users.add(user);

        Hawk.put(Constant.USERLIST, Users);
    }

    public final void addOrUpdateUser(@NotNull UserInfo newuser) {
        if (newuser == null) {
            return;
        }
        UserInfo oldUser = this.findUser(newuser);
        if (oldUser != null) {
            Users.set(oldUser.Id-1, newuser);
            Log.d("aliT", newuser.toString());
            Hawk.put(Constant.USERLIST, Users);
        } else {
            this.addUser(newuser);
        }

    }

    public final void signIn(@NotNull UserInfo currentUser) {
        if (currentUser == null) {
            return;
        }
        Hawk.put(Constant.CURRENTUSER, currentUser);
    }

    public final void signOut() {
        Hawk.delete(Constant.CURRENTUSER);
    }

    public final UserInfo getCurrentUser() {
        if (Hawk.contains(Constant.CURRENTUSER)) {
            return Hawk.get(Constant.CURRENTUSER);
        } else
            return null;
    }

    @Nullable
    public final UserInfo findUser(@NotNull UserInfo user) {
        if (user == null) {
            return null;
        }
        List<UserInfo> users = this.getUsers();
        for (UserInfo it : users) {
            if (it.Id==user.Id)
                return it;
        }
        return null;
    }

    private final void init() {

//        Hawk.deleteAll();
        List<UserInfo> users = this.getUsers();
        if (users.size() <= 0) {
            Users = new ArrayList<UserInfo>();
            UserInfo admin = new UserInfo();
            admin.FirstName = "Ali";
            admin.Family = "Tarighi";
            admin.Mobile = 1;
            admin.Id = 1;
            admin.City = "Tehran";
            this.addUser(admin);
        }
    }


}

File: app/src/main/java/com/tarighi/register/UserController.kt
/*
package com.tarighi.register

import com.orhanobut.hawk.Hawk
import java.util.ArrayList

 object UserController {
    @JvmStatic
    private var Users: MutableList<UserInfo>? = null
    fun getUsers(): List<UserInfo>? {
        Users = Hawk.get<MutableList<UserInfo>>(Constant.USERLIST);
        if(this.Users==null)
            this.Users = ArrayList();
        return this.Users;
    }

    private fun setUsers(users: MutableList<UserInfo>) {
        Users = users
    }

    private fun addUser(user: UserInfo) {
        Users!!.add(user)
        Hawk.put(Constant.USERLIST, Users);
    }

//    fun getUser(UserName: String): UserInfo {
//        return UserInfo()
//        //        return Users.stream().filter(x-> x.UserName.equalsIgnoreCase(UserName)).findFirst().get();
//    }

    fun addOrUpdateUser(newuser: UserInfo) {

        var oldUser=findUser(newuser);
        if(oldUser!=null)
        {
            oldUser=newuser;
            Hawk.put(Constant.USERLIST, Users);
        }
        else
        {
            addUser(newuser);
        }
    }
    fun signIn(currentUser: UserInfo) {
        Hawk.put(Constant.CURRENTUSER, currentUser);
    }
    fun signOut() {
        Hawk.delete(Constant.CURRENTUSER);
    }

    fun findUser( user: UserInfo): UserInfo? {
        var list = getUsers();
        list!!.forEach {
            if (it.equals(user)) {
                return it;

            }
        }
        return null
    }


    private fun init() {
        if (getUsers()!!.size > 0)
            return

        Users = ArrayList();
        var admin = UserInfo();
        admin.FirstName = "Ali";
        admin.Family = "Tarighi";
        admin.Mobile = 1;

        addUser(admin);

    }

}
*/
